cmake/alpakaCommon.cmake

```diff
@@ -313,257 +313,146 @@ endif()
-                    message(FATAL_ERROR "Clang as a CUDA compiler does not support OpenMP 5!")
-                endif()
-
-                foreach(_CUDA_ARCH_ELEM ${ALPAKA_CUDA_ARCH})
-                    target_compile_options(alpaka INTERFACE  "--cuda-gpu-arch=sm_${_CUDA_ARCH_ELEM}")
-                endforeach()
+    if(CMAKE_CUDA_COMPILER)
+        enable_language(CUDA)
+        find_package(CUDAToolkit REQUIRED)
 
-                target_compile_options(alpaka INTERFACE "--cuda-path=${CUDA_TOOLKIT_ROOT_DIR}")
+        if(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE)
+            message(FATAL_ERROR "CUDA cannot be used together with Boost.Fiber!")
+        endif()
 
-                # This flag silences the warning produced by the Dummy.cpp files:
-                # clang: warning: argument unused during compilation: '--cuda-gpu-arch=sm_XX'
-                # This seems to be a false positive as all flags are 'unused' for an empty file.
-                target_compile_options(alpaka INTERFACE "-Qunused-arguments")
+        target_compile_features(alpaka INTERFACE cuda_std_${ALPAKA_CXX_STANDARD})
 
-                # Silences warnings that are produced by boost because clang is not correctly identified.
-                target_compile_options(alpaka INTERFACE "-Wno-unused-local-typedef")
+        alpaka_compiler_option(CUDA_SHOW_REGISTER "Show kernel registers and create device ASM" DEFAULT)
+        alpaka_compiler_option(CUDA_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps 'CMakeFiles/<targetname>.dir'" DEFAULT)
+        alpaka_compiler_option(CUDA_EXPT_EXTENDED_LAMBDA "Enable experimental, extended host-device lambdas in CUDA with nvcc" ON)
 
-                if(ALPAKA_FAST_MATH STREQUAL ON)
-                    # -ffp-contract=fast enables the usage of FMA
-                    alpaka_set_compiler_options(HOST_DEVICE target alpaka "-ffast-math" "-ffp-contract=fast")
-                endif()
+        if(CMAKE_CUDA_COMPILER_ID STREQUAL "Clang")
+            message(STATUS "clang is used as CUDA compiler")
 
-                if(ALPAKA_FTZ STREQUAL ON)
-                    alpaka_set_compiler_options(HOST_DEVICE target alpaka "-fcuda-flush-denormals-to-zero")
-                endif()
+            if(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE OR ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE)
+                message(FATAL_ERROR "Clang as a CUDA compiler does not support OpenMP 2!")
+            endif()
+            if(ALPAKA_ACC_ANY_BT_OMP5_ENABLE)
+                message(FATAL_ERROR "Clang as a CUDA compiler does not support OpenMP 5!")
+            endif()
 
-                if(ALPAKA_CUDA_SHOW_REGISTER STREQUAL ON)
-                    alpaka_set_compiler_options(DEVICE target alpaka "-Xcuda-ptxas=-v")
-                endif()
+            # libstdc++ since version 7 when GNU extensions are enabled (e.g. -std=gnu++11)
+            # use `__CUDACC__` to avoid defining overloads using non-standard `__float128`.
```

Comment:
There is a `s` missing here (`use` -> `uses`).

Reply:
fixed